### ISSUE TITLE
Setting time and date display format in rtcr clientraw.txt

### ISSUE THREAD

**Issue by ktrue** [gjr80/weewx-saratoga#4]
Since the AJAX scripts don't do much other than display the contents of time fields, I would be really helpful for them to be displayed uniformly for fields 32 (with seconds) and fields 134,135,166,174,175 with hours:minutes.

The preferred format for USA is '%-I:%M:%S_%p' for field 32 and '%-I:%M_%p' for the other ones.
For non-USA, it's %H:%M:%S' for field 32 and '%H:%M' for the other ones.

Could that be a setting in the [RealtimeClientraw] area or just automatically detected if the temperature unit is F (or != C)?

Likewise, the date format for field 74 should be '%-m/%-d/%Y' for USA and '%-d/%-m/%Y' otherwise.

**Comment by gjr80** [comment]
Agree, just need to find the best way/place to set the format.One thing I keep in the back of my mind is trying to minimise the number of places where we set formats and if at all possible avoid (essentially) setting the same thing in two different places.

Thinking aloud, as there are no other similar fields amongst the clientraws this is unique to `clientraw.txt`. `WEEWXtags.php` has numerous 'time' tags, but if I am correct the `timeFormat` and `timeOnlyFormat` settings in `Settings.php` control how times from `WEEWXtags.php` are presented when the dashboard is first rendered by php/html. The formats in `clientraw.txt` essentially determine the format of the same dashboard fields when updated by javascript. If I understand correctly it would seem there needs to be some coordination between the `timeFormat` and `timeOnlyFormat` settings in `Settings.php` and the formats used for the `clientraw.txt` fields listed above to get consistent time presentation on the dashboard.

Still thinking aloud, we could infer the time formats from the likes of temperature units but as night follows day there will be someone who wants Fahrenheit with 24 hour time or Celsius with AM/PM (or some other format). So I think we are best served explicitly setting the formats. 

So at present, given there is no need to use the same formats in `WEEWXtags.php`, I am leaning to setting these formats explicitly under `[WeewxSaratoga] [[RealtimeClientraw]]` with say `long_time_format` and `short_time_format` config options (we could use `timeFormat` and `timeOnlyFormat` as in `Settings.php` but since they will have different settings I think that would only confuse people).

**Comment by ktrue** [comment]
I'm OK with having entries in [[RealtimeClient]] with the long_time_format and short_time_format config options.  Both flavors of the time formats (USA and 'standard') should appear, with the USA ones commented out so it's an easy 'remove the #' to change the format.  AFAIK, those are the two basic formats provided by WD (12 or 24hr time, with or without seconds).  Only the time used with the station name uses seconds, the others are all just hours:minutes.

**Comment by gjr80** [comment]
Well the good news is the date-time formats in realtime clientraw are implemented and working, though we have a bit of an issue with the config file parser library (configobj) and comments. As mentioned there are a few peculiarities, one is that we cannot have a comment line as the last line of a given level, that's fine I thought I could work around that with a `[RealtimeClientraw]` stanza like:
```
    [[RealtimeClientraw]]
        remote_server_url = something
        min_interval = 10

        # Python date-time format strings. Format string codes as per 
        # https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
     
        # Date format. Uncomment one line.
        # date_format = %-m/%-d/%Y  # recommended for USA users
        date_format = %-d/%-m/%Y  # recommended for non-USA users

        # Long format times (HMS). Uncomment one line.
        # long_time_format = %-I:%M:%S_%p  # recommended for USA users
        long_time_format = %H:%M:%S  # recommended for non-USA users
    
        # Short format times (HM). Uncomment one line.
        # short_time_format = %-I:%M_%p  # recommended for USA users
        short_time_format = %H:%M  # recommended for non-USA users
```

Unfortunately configobj doesn't like a line that ends with a comment so it just drops the trailing comment. The resulting config stanza looks like:
```
    [[RealtimeClientraw]]
        remote_server_url = something
        min_interval = 10

        # Python date-time format strings. Format string codes as per 
        # https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
     
        # Date format. Uncomment one line.
        # date_format = %-m/%-d/%Y  # recommended for USA users
        date_format = %-d/%-m/%Y

        # Long format times (HMS). Uncomment one line.
        # long_time_format = %-I:%M:%S_%p  # recommended for USA users
        long_time_format = %H:%M:%S
    
        # Short format times (HM). Uncomment one line.
        # short_time_format = %-I:%M_%p  # recommended for USA users
        short_time_format = %H:%M
```

I realise it is a little different to the approach used in the Saratoga settings files but I was going to suggest something along these lines:
```
    [[RealtimeClientraw]]
        remote_server_url = something
        min_interval = 10

        # Python date-time format strings. Format string codes as per 
        # https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
         
        # Date format. Recommended entries are:
        #   date_format = %-m/%-d/%Y  # recommended for USA users
        #   date_format = %-d/%-m/%Y  # recommended for non-USA users
        date_format = %-d/%-m/%Y

        # Long format times (HMS). Recommended entries are:
        #   long_time_format = %-I:%M:%S_%p  # recommended for USA users
        #   long_time_format = %H:%M:%S  # recommended for non-USA users 
        long_time_format = %H:%M:%S
        
        # Short format times (HM). Recommended entries are:
        #   short_time_format = %-I:%M_%p  # recommended for USA users
        #   short_time_format = %H:%M  # recommended for non-USA users
        short_time_format = %H:%M
```

This is more along the lines of how we do option choices in WeeWX (probably because of configobj limitations), more of a 'change the setting' approach rather than 'comment/uncomment' lines.

**Comment by ktrue** [comment]
No issue with the different method of specification necessitated by the configobj limitations.  Your example above looks good.

I'd suggest mods to the remote_server_url and min_interval entries as:

```
    [[RealtimeClientraw]]
       # min_interval sets the generation frequency - default is every 10 seconds for a new clientraw.txt
        min_interval = 10

       #  If using an external website, configure remote_server_url to point to the post_clientraw.php script
       #  on your website like:
       #   remote_server_url = http://your.website.com/post_clientraw.php
       #
       # To use the webserver on this system, leave the next entry commented out
       # remote_server_url = http://your.website.com/post_clientraw.php
```

**Comment by ktrue** [comment]
I see you've already done the exposition in install.py that I'd suggested above.  Bravo! 
(I like your phrasing better than the ones I suggested above).

**Comment by ktrue** [comment]
This appears to be fixed in fed7915 so I'm closing this.  Thanks!
